spyder/plugins/pylint/main_widget.py

```diff
@@ -0,0 +1,899 @@
+
+                for ext in ('.py', '.pyw'):
+                    if osp.isfile(modname+ext):
+                        modname = modname + ext
+                        break
+
+                if osp.isdir(self.filename):
+                    parent = modules.get(modname)
+                    if parent is None:
+                        item = QTreeWidgetItem(title_item, [module],
+                                               QTreeWidgetItem.Type)
+                        item.setIcon(0, ima.icon('python'))
+                        modules[modname] = item
+                        parent = item
+                else:
+                    parent = title_item
+
+                if len(msg_id) > 1:
+                    if not message_name:
+                        message_string = "{msg_id} "
+                    else:
+                        message_string = "{msg_id} ({message_name}) "
+
+                message_string += "line {lineno}: {message}"
+                message_string = message_string.format(
+                    msg_id=msg_id, message_name=message_name,
+                    lineno=lineno, message=message)
+                msg_item = QTreeWidgetItem(
+                    parent, [message_string], QTreeWidgetItem.Type)
+                msg_item.setIcon(0, ima.icon('arrow'))
+                self.data[id(msg_item)] = (modname, lineno)
+
+
+class PylintWidget(PluginMainWidget):
+    """
+    Pylint widget.
+    """
+    DEFAULT_OPTIONS = {
+        'history_filenames': [],
+        'max_entries': 30,
+        'project_dir': None,
+    }
+
+    DATAPATH = get_conf_path('pylint.results')
+    VERSION = '1.1.0'
+
+    # --- Signals
+    sig_edit_goto_requested = Signal(str, int, str)
+    """
+    This signal will request to open a file in a given row and column
+    using a code editor.
+
+    Parameters
+    ----------
+    path: str
+        Path to file.
+    row: int
+        Cursor starting row position.
+    word: str
+        Word to select on given row.
+    """
+
+    sig_start_analysis_requested = Signal()
+    """
+    This signal will request the plugin to start the analysis. This is to be
+    able to interact with other plugins, which can only be done at the plugin
+    level.
+    """
+
+    def __init__(self, name=None, plugin=None, parent=None,
+                 options=DEFAULT_OPTIONS):
+        super().__init__(name, plugin, parent, options)
+
+        # Attributes
+        self._process = None
+        self.output = None
+        self.error_output = None
+        self.filename = None
+        self.rdata = []
+        self.curr_filenames = self.get_option('history_filenames')
+        self.code_analysis_action = None
+        self.browse_action = None
+
+        # Widgets
+        self.filecombo = PythonModulesComboBox(self)
+        self.ratelabel = QLabel(self)
+        self.datelabel = QLabel(self)
+        self.treewidget = ResultsTree(self)
+
+        # FIXME: Move to plugin?

```

Comment:
So, should we move the things below this line to the plugin? (or was this comment before the move to the API maybe?)

Reply:
Removed the FIXME notice.